drivers/client_location_history/app/controllers/client_location_history/clients_controller.rb

```diff
@@ -15,7 +15,11 @@ class ClientsController < ApplicationController
     before_action :set_client
 
     def map
-      @locations = @client.client_location_histories.where(located_on: filter.range)
+      client_ids = [@client.id]
+      # Include any source clients with a location, but make sure we only bring in those that are visible to the current user
+      client_ids << ::GrdaWarehouse::Hud::Client.source_visible_to(current_user).where(id: @client.source_client_ids).pluck(:id) if @client.destination?
```

Comment:
I think you want to hint with the client ids, that should help performance.
Also, += means you don't need to flatten later.
```suggestion
      client_ids += ::GrdaWarehouse::Hud::Client.source_visible_to(current_user, client_ids: @client.source_client_ids).where(id: @client.source_client_ids).pluck(:id) if @client.destination?
```

Reply:
included this [here](https://github.com/greenriver/hmis-warehouse/pull/5132/commits/b19cd431c741fa2a5620250a0df364730fc2c72d)